VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift

```diff
@@ -0,0 +1,356 @@
+    {
+        let sessionManager = sessionManager as! VimeoSessionManager
+        let responseSerializer = sessionManager.responseSerializer as! VimeoResponseSerializer
+        
+        do
+        {
+            switch self.currentRequest
+            {
+            case .CreateVideo:
+                self.uploadTicket = try responseSerializer.processCreateVideoResponse(task.response, url: url, error: error)
+                
+            case .UploadVideo:
+                break
+                
+            case .ActivateVideo:
+                self.videoUri = try responseSerializer.processActivateVideoResponse(task.response, url: url, error: error)
+                
+            case .VideoSettings:
+                self.video = try responseSerializer.processVideoSettingsResponse(task.response, url: url, error: error)
+                
+            case .CreateThumbnail:
+                self.thumbnailTicket = try responseSerializer.processCreateThumbnailResponse(task.response, url: url, error: error)
+                
+            case .UploadThumbnail:
+                break
+                
+            case .ActivateThumbnail:
+                self.thumbnail = try responseSerializer.processActivateThumbnailResponse(task.response, url: url, error: error)
+            }
+        }
+        catch let error as NSError
+        {
+            self.error = error
+            self.currentTaskIdentifier = nil
+            self.state = .Finished
+        }
+        
+        return nil
+    }
+    
+    override func taskDidComplete(sessionManager sessionManager: AFURLSessionManager, task: NSURLSessionTask, error: NSError?)
+    {
+        if self.currentRequest == .UploadVideo
+        {
+            NSFileManager.defaultManager().deleteFileAtURL(self.videoUrl)
+        }
+        else if self.currentRequest == .UploadThumbnail
+        {
+            if let thumbnailFileUrl = self.thumbnailFileUrl {
+                NSFileManager.defaultManager().deleteFileAtURL(thumbnailFileUrl)
+            }
+        }
+        
+        if self.error == nil
+        {
+            if let taskError = task.error // task.error is reserved for client-side errors, so check it first
+            {
+                let domain = self.errorDomainForRequest(self.currentRequest)
+                self.error = taskError.errorByAddingDomain(domain)
+            }
+            else if let error = error
+            {
+                let domain = self.errorDomainForRequest(self.currentRequest)
+                self.error = error.errorByAddingDomain(domain)
+            }
+        }
+        
+        var nextRequest = CAMUploadRequest.nextRequest(self.currentRequest)
+        if self.error != nil || nextRequest == nil
+        {
+            self.currentTaskIdentifier = nil
+            self.state = .Finished
+            
+            return
+        }
+        else if nextRequest == .VideoSettings && self.videoSettings == nil
```

Comment:
This method is a bit messy because it basically contains the logic for a state machine and has to handle setting up the different states and transitions.  There are two states which have two different transitions that can occur based on whether or not the descriptor was given certain inputs (video settings and thumbnail image).  Ideally, we should add an actual state machine class to the descriptors which will allow us to encapsulate all of this logic and clean up the code.  For now, I cleaned up the code paths as best I could and added comments for clarity.